crates/storage/src/kv_store.rs

```diff
@@ -17,13 +17,8 @@ use core::ops::Deref;
 
 /// The key of the storage.
 pub type Key = Vec<u8>;
-#[cfg(feature = "std")]
 /// The value of the storage. It is wrapped into the `Arc` to provide less cloning of massive objects.
-pub type Value = std::sync::Arc<Vec<u8>>;
-
-#[cfg(not(feature = "std"))]
-/// The value of the storage. It is wrapped into the `Rc` to provide less cloning of massive objects.
-pub type Value = alloc::rc::Rc<Vec<u8>>;
+pub type Value = alloc::sync::Arc<Vec<u8>>;
```

Comment:
is this change intended to be in this PR? if so, should we use `alloc::sync::Arc` from this point onwards instead of stdlib?

Reply:
`alloc::sync::Arc` and `std::sync::Arc` are the same in the context of `std`. Yes, without this change, it is not possible to compile code because I require `Send` on all `BoxedStreams` now